backend/test/users_test.js

```diff
@@ -734,3 +737,101 @@ describe("Test get student completed courses", () => {
   });
 });
 
+describe("Testing Login", () => {
+  before(async () => {
+    await users.insertStudentUser(
+      "team",
+      "primus1234",
+      "testing@gmail.com",
+      250502459
+    );
+  });
+
+  it(`Empty username responds with ${expectedUsernameNotEmpty}`, function(done) {
+    users.login(null, "primus1234").catch(error => {
```

Comment:
Is null username the same as empty username? Like if you put in "" versus null, would it behave the same?

Reply:
I think it wouldn't behave the same since only null are caught in the validation

Reply:
Good catch, "" != null, the condition checks for 
userInfo == undefined || userInfo.length == 0
so it will catch it. I will add in a test for empty instead of null